docs/source/design/authoring.rst

```diff
@@ -127,19 +127,19 @@ Let's assume we have a workflow like ::
         d = t2(a=y, b=b)
         return x, d
 
-As discussed in the Promise object's documentation, when a task is called from inside a workflow, the Python native values that the raw underlying functions return are first converted into Flyte IDL literals, and then wrapped inside ``Promise`` objects. One ``Promise`` is created for every return variable.
+As discussed in the Promise object's documentation, when a task is called from inside a workflow, the Python native values returned by the raw underlying functions are first converted into Flyte IDL literals, and then wrapped inside ``Promise`` objects. One ``Promise`` is created for every return variable.
 
-When the next task is called, logic is triggered to unwrap these promises.
+When the next task is called, the logic is triggered to unwrap these Promises.
 
 Compilation
 ===========
-When a workflow is compiled, instead of producing ``Promise`` objects that wrap literal values, they wrap a :py:class:`flytekit.core.promise.NodeOutput` instead. This is how data dependency is tracked between tasks.
+When a workflow is compiled, instead of producing Promise objects that wrap literal values, they wrap a :py:class:`flytekit.core.promise.NodeOutput` instead. This helps track data dependency between tasks.
 
 Branch Skip
 ===========
-If it's been determined that a conditional is not true, then Flytekit will skip calling the task. This way, any side-effects in the task logic will not be run.
+If a conditional is determined to be false, then Flytekit will skip calling the task. This avoids running the faulty task logic.
```

Comment:
```suggestion
If a :py:func:`flytekit.conditional` is determined to be false, then Flytekit will skip calling the task. This avoids running the unintended task.
```

Reply:
This reference doesn't seem to work. Can you tell me the file you are trying to refer to?  This-> GitHub/prod/flytekit/flytekit/core/condition.py ? @samhita-alla